sympy/polys/numberfields/modules.py

```diff
@@ -0,0 +1,1794 @@
+
+    def reduced_mod_p(self, p):
+        """
+        Produce a version of this :py:class:`~.ModuleElement` in which all
+        numerator coefficients have been reduced mod *p*.
+        """
+        return make_mod_elt(self.module,
+                            self.col.convert_to(FF(p)).convert_to(ZZ),
+                            denom=self.denom)
+
+    @classmethod
+    def from_int_list(cls, module, coeffs, denom=1):
+        """
+        Make a :py:class:`~.ModuleElement` from a list of ints (instead of a
+        column vector).
+        """
+        col = to_col(coeffs)
+        return cls(module, col, denom=denom)
+
+    @property
+    def n(self):
+        """The length of this element's column."""
+        return self.module.n
+
+    def __len__(self):
+        return self.n
+
+    def column(self, domain=None):
+        """
+        Get a copy of this element's column, optionally converting to a domain.
+        """
+        return self.col.convert_to(domain)
+
+    @property
+    def coeffs(self):
+        return self.col.flat()
+
+    @property
+    def QQ_col(self):
+        """
+        :py:class:`~.DomainMatrix` over :ref:`QQ`, equal to
+        ``self.col / self.denom``, and guaranteed to be dense.
+
+        See Also
+        ========
+
+        .Submodule.QQ_matrix
+
+        """
+        if self._QQ_col is None:
+            self._QQ_col = (self.col / self.denom).to_dense()
+        return self._QQ_col
+
+    def to_parent(self):
+        """
+        Transform into a :py:class:`~.ModuleElement` belonging to the parent of
+        this element's module.
+        """
+        if not isinstance(self.module, Submodule):
+            raise ValueError('Not an element of a Submodule.')
+        return make_mod_elt(
+            self.module.parent, self.module.matrix * self.col,
+            denom=self.module.denom * self.denom)
+
+    def to_ancestor(self, anc):
+        """
+        Transform into a :py:class:`~.ModuleElement` belonging to a given
+        ancestor of this element's module.
+
+        Parameters
+        ==========
+
+        anc : :py:class:`~.Module`
+
+        """
+        if anc == self.module:
+            return self
+        else:
+            return self.to_parent().to_ancestor(anc)
+
+    def over_power_basis(self):
+        """
+        Transform into a :py:class:`~.PowerBasisElement` over our
+        :py:class:`~.PowerBasis` ancestor.
+        """
+        e = self
+        while not isinstance(e.module, PowerBasis):
+            e = e.to_parent()
+        return e
+
+    def is_compat(self, other):
+        """
+        Test whether other is another :py:class:`~.ModuleElement` with same
+        module.
+        """
+        return isinstance(other, ModuleElement) and other.module == self.module
+
+    def make_compat(self, other):
+        """
+        Try to make a compatible pair of :py:class:`~.ModuleElement`, one
+        equivalent to this one, and one equivalent to the other.
```

Comment:
In other contexts in the polys domains this is referred to as unification and the method would be called `unify`. Maybe it would make sense to use that name here or maybe it would be confusing...